### PR TITLE
Redid OI, still need updating on diagram, and included commands for Crabarms to flap out.

### PR DESCRIPTION
**Pull request by pak-plau**

### CODE REVIEW

File: src/org/usfirst/frc/team694/robot/commands/BITCOINCheckCommand.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
 public class BITCOINCheckCommand extends ConditionalCommand { 
     
     public BITCOINCheckCommand() {
-        super(new BITCOINAutoCommand(), new BITCOINManualCommand());
+        super(new BITCOINAutoCommand(), new CrabArrowAcquireCommand());
```
Comment (PrathamRawat) [inline src/org/usfirst/frc/team694/robot/commands/BITCOINCheckCommand.java:10]: This change should not be made in the OI branch. You should talk to the people working on BITCOIN and tell them why this change is necessary, and have them make it, they know the code better and they should implement it themselves.

File: src/org/usfirst/frc/team694/robot/commands/CrabArmDeacquireCommand.java [hunk 2]
```diff
@@ -14,7 +14,7 @@ protected void initialize() {
     }
 
     protected void execute() {
-        Robot.crabArm.deacquire();
+        Robot.crabArm.flapOut();
```
Comment (PrathamRawat) [inline src/org/usfirst/frc/team694/robot/commands/CrabArmDeacquireCommand.java:17]: Flapping out should probably not be running at full speed, but the name change is welcome.

File: src/org/usfirst/frc/team694/robot/commands/CrabArmFlapAndSpatulaFlipDownCommand.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ */
+public class CrabArmFlapAndSpatulaFlipDownCommand extends CommandGroup {
+    private static final int timeDelayForSpatulaFlip = 2;
+    public CrabArmFlapAndSpatulaFlipDownCommand() {
+        addSequential(new CrabArmFlapOutCommand());
+        addSequential(new WaitCommand(timeDelayForSpatulaFlip));
+        addSequential(new SpatulaFlipDownCommand());
+        addSequential(new WaitCommand(timeDelayForSpatulaFlip));
+        addSequential(new CrabArmStopCommand());
+    }
```
Comment (PrathamRawat) [inline src/org/usfirst/frc/team694/robot/commands/CrabArmFlapAndSpatulaFlipDownCommand.java]: This logic is probably flawed.

File: src/org/usfirst/frc/team694/robot/commands/CrabArmFlapAndSpatulaFlipDownCommand.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+import edu.wpi.first.wpilibj.command.CommandGroup;
+import edu.wpi.first.wpilibj.command.TimedCommand;
+import edu.wpi.first.wpilibj.command.WaitCommand;
+
+/**
+ *
+ */
+public class CrabArmFlapAndSpatulaFlipDownCommand extends CommandGroup {
+    private static final int timeDelayForSpatulaFlip = 2;
```
Comment (PrathamRawat) [inline src/org/usfirst/frc/team694/robot/commands/CrabArmFlapAndSpatulaFlipDownCommand.java]: This should be more specific in its name, and at least reference the units of time being used.

File: src/org/usfirst/frc/team694/robot/commands/CrabArmFlapAndSpatulaFlipUpCommand.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import org.usfirst.frc.team694.robot.Robot;
+
+import edu.wpi.first.wpilibj.command.CommandGroup;
+import edu.wpi.first.wpilibj.command.WaitCommand;
+
+/**
+ *
+ */
+public class CrabArmFlapAndSpatulaFlipUpCommand extends CommandGroup {
+private static final int timeDelayForSpatulaFlip = 2;
```
Comment (PrathamRawat) [inline src/org/usfirst/frc/team694/robot/commands/CrabArmFlapAndSpatulaFlipUpCommand.java]: Same comments for the Flip-Down command.

File: src/org/usfirst/frc/team694/robot/commands/CrabArmFlapOutCommand.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+ */
+public class CrabArmFlapOutCommand extends Command {
+
+    public CrabArmFlapOutCommand() {
+        requires(Robot.crabArm);
+    }
+
+    protected void execute() {
+        Robot.crabArm.flapOut();
+    }
```
Comment (PrathamRawat) [inline src/org/usfirst/frc/team694/robot/commands/CrabArmFlapOutCommand.java]: This is definitely not going to work. There are quite a few things, but the biggest one is that this will run the motors in the opposite direction at full speed forever and stall out the motors or break the crab-arms.

File: src/org/usfirst/frc/team694/robot/commands/SpatulaFlipToggleCommand.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+public class SpatulaFlipToggleCommand extends InstantCommand {
+
+    public SpatulaFlipToggleCommand() {
+        super();
+        requires(Robot.spatula);
+    }
+    
+    protected void initialize() {
+        Robot.spatula.toggle();
+    }
```
Comment (PrathamRawat) [inline src/org/usfirst/frc/team694/robot/commands/SpatulaFlipToggleCommand.java:19]: What is the point of this? Toggling the spatula should involve the Crab Arms, this needs to add that functionality too.

File: src/org/usfirst/frc/team694/robot/commands/auton/SameSideRightAngleAutonCommand.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import org.usfirst.frc.team694.robot.commands.DrivetrainRotateDegreesPIDCommand;
+import org.usfirst.frc.team694.robot.commands.GrabberOpenCommand;
+import org.usfirst.frc.team694.robot.commands.LiftMaxUpCommand;
+
+import edu.wpi.first.wpilibj.command.CommandGroup;
+
+/**
+ *
+ */
+public class SameSideRightAngleAutonCommand extends CommandGroup {
```
Comment (PrathamRawat) [inline src/org/usfirst/frc/team694/robot/commands/auton/SameSideRightAngleAutonCommand.java:13]: This branch should not be including these Auton Commands, make sure you pull before you push.

File: src/org/usfirst/frc/team694/robot/subsystems/Spatula.java [hunk 9]
```diff
@@ -91,4 +91,11 @@ public void loosenCubeGrip() {
     public boolean isCubeDetected() {
         return limitSwitch.get();
     }
+    public void toggle() {
```
Comment (PrathamRawat) [inline src/org/usfirst/frc/team694/robot/subsystems/Spatula.java:94]: Yes, the toggle command definitely needs to add the flap out functionality.

**Review by PrathamRawat — CHANGES_REQUESTED** [review]
This has a lot of problems, the first of which is that these changes should not really be on this branch. I suggest you remove all of the changes you make which do not pertain to OI from your commit, and only include the changes which matter to this branch. Delete all of your changes to the Crab Arms, as that logic is most definitely flawed.

**Review by TacoTechnica — APPROVED** [review]
Problems still exist, but we'll figure them out while doing a println robot

**Review by PrathamRawat — APPROVED** [review]
